next_sparseconvnet/networks/architectures.py

```diff
@@ -0,0 +1,60 @@
+
+import torch
+import sparseconvnet as scn
+from .building_blocks import ResidualBlock_downsample, ResidualBlock_basic, ResidualBlock_upsample
+
+class UNet(torch.nn.Module):
+        def __init__(self, spatial_size, inplanes, kernel, stride, basic_num, level_depth, dim = 3):
+            torch.nn.Module.__init__(self)
+
+            self.basic_num = basic_num
+            self.level_depth = level_depth
+
+            self.inp = scn.InputLayer(dim, spatial_size)
+
+            #Creo las layers de la estructura en U con un par de bucles
+
+            self.downsample = []
+            self.upsample = []
+            self.basic_down = []
+            self.basic_up = []
+
+            for i in range(level_depth):
+                for j in range(basic_num):
+                    if i != level_depth - 1: #es decir, mientras no llegue al nivel final, hacer 3 basic en ambos lados
+                        self.basic_down.append(ResidualBlock_basic(inplanes, kernel))
```

Comment:
I think it would be better to somehow separate basic blocks with different size, ie something of a style;
```
self.basic_down = [[] for i in range(level_depth-1)] #initial to  level_depth-1 empty lists
self.bottleneck = [] #lowest layer shared between 2 branches
self.basic_up = [[] for i in range(level_depth-1)] #initial to  level_depth-1 empty lists
```
then loops go as 
```
for i in range(level_depth-1):
    for j in range(basic_num):
        self.basic_down[i].append(ResidualBlock_basic...)
        self.basic_up[i].append(...)
    self.downsample.append(ResidualBlock_downsample...)
    self.upsample.append(....)
    inplanes *=2
#create last layer    
for j in range(basic_num):
    self.bottleneck.append(ResidualBlock_basic...)
    
```